src/BindPattern.jl

```diff
@@ -294,10 +309,15 @@ function bind_pattern!(
             end
 
             field_type = nothing
-            for (fname, ftype) in zip(Base.fieldnames(bound_type), Base.fieldtypes(bound_type))
-                if fname == field_name
-                    field_type = ftype
-                    break
+            if field_name == fieldnames(Symbol)[1]
+                # special case Symbol's hypothetical name field.
+                field_type = String
```

Comment:
what's the thinking on the supposed inner type of a `Symbol` being a `String` vs a `Symbol`? What use cases does this unlock?

Reply:
If the inner type of a `Symbol` were itself a `Symbol`, then there wouldn't be a point of ever accessing the inner member (it would be the same as the container).  With it being a `String`, you can look for symbols by name.